### PR TITLE
Modelreposync

### PR DESCRIPTION
**Pull request by davidlane-iteratec**
https://github.com/IndustryFusion/DigitalTwin/issues/49

### CODE REVIEW

File: fusionbackend/src/main/resources/config/application.yaml [hunk 1]
```diff
@@ -48,5 +48,10 @@ logging:
   level:
     root: INFO
 
+fusionbackend:
+  modelreposyncactive: true
```
Comment (jsylupp) [inline fusionbackend/src/main/resources/config/application.yaml]: kebab case for consistency and readability?

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetService.java [hunk 2]
```diff
@@ -503,6 +502,7 @@ public String getAssetByIdAsNgsiLD(Long assetId) throws IOException {
             assetDto.setCompanyId(null);
             assetDto.setRoom(null);
             assetDto.setRoomId(null);
+            assetDto.setSubsystemIds(null);
```
Comment (jsylupp) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetService.java]: I think subsystemIds are needed here as this is the only reference to subsystems (there is no setSubsystems())

File: fusionfrontend/src/app/core/services/api/if-api.service.ts [hunk 3]
```diff
@@ -23,18 +23,29 @@ import { Observable } from 'rxjs';
...
+  showResult: boolean;
 
   constructor(private http: HttpClient) {
   }
 
-  getExportLinkEcosystemManager(asOwl: boolean): string {
-    let path = `/zipexport`;
-    if (asOwl) {
-      path = '/owlexport';
+  getExportLinkEcosystemManager(exportType: string): string {
```
Comment (jsylupp) [inline fusionfrontend/src/app/core/services/api/if-api.service.ts:35]: consider enum

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/export/FleetManagerImportExportService.java [hunk 4]
```diff
@@ -114,19 +145,20 @@ public ImportResult tryImportOfZipEntry(final ZipEntry entry,
                                             final Long factorySiteId,
                                             final ImportResult prevImportResult) throws IOException {
         boolean isEntryImported = entry != null && zipInputStream != null;
-        int totalEntitySkippedCount = prevImportResult.totalEntitySkippedCount;
+        int totalEntitySkippedCount = prevImportResult.getTotalEntitySkippedCount();
```
Comment (jsylupp) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/export/FleetManagerImportExportService.java:148]: Shouldn't ProcessingResultDto also be used here?

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/export/EcosystemManagerImportExportService.java [hunk 5]
```diff
@@ -128,4 +159,183 @@ public ImportResult tryImportOfZipEntry(final ZipEntry entry,
...
+        final String filename = URLEncoder.encode(assetTypeTemplate.getAssetType().getLabel(), StandardCharsets.UTF_8)
+                + "-V" + assetTypeTemplate.getPublishedVersion();
+        return getAttFilePath(filename, extension);
+    }
+
+    private Path getAttFilePath(final String filename, final String extension) {
+        return getAttDirPath().resolve(filename + extension);
+    }
+
+    private Path getEcomanFilePath(final String filename, final String extension) {
```
Comment (jsylupp) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/export/EcosystemManagerImportExportService.java]: optional: consider full name EcosystemManager (same below and in line 241)

**Review by jsylupp — CHANGES_REQUESTED** [review]
Well done!

File: fusionbackend/src/main/resources/config/application.yaml (hunk 1 above)
Comment (davidlane-iteratec) [reply fusionbackend/src/main/resources/config/application.yaml]: ok

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/ModelRepoSyncService.java [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+            log.warn("Could not create local model repo path {}: {}", localGitPath, e);
+            return false;
+        }
+        return checkoutRepo(localGitPath);
+    }
+
+    private boolean checkoutRepo(Path localGitPath) {
+        if (!isGitRepo(localGitPath)) {
+            try {
+                final boolean isDirectoryEmpty = isDirectoryEmpty(localGitPath);
```
Comment (jsylupp) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/ModelRepoSyncService.java:99]: consider extracting methods

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetService.java (hunk 2 above)
Comment (davidlane-iteratec) [reply fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetService.java]: Ok

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/export/FleetManagerImportExportService.java (hunk 4 above)
Comment (davidlane-iteratec) [reply fusionbackend/src/main/java/io/fusion/fusionbackend/service/export/FleetManagerImportExportService.java:148]: I considered replaying ImportResult everywhere with ProcessingResultDto but rejected it in favour of a refactoring task. ImportResult is used also in Frontend.

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/export/EcosystemManagerImportExportService.java (hunk 5 above)
Comment (davidlane-iteratec) [reply fusionbackend/src/main/java/io/fusion/fusionbackend/service/export/EcosystemManagerImportExportService.java]: Ok

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/ModelRepoSyncService.java (hunk 6 above)
Comment (davidlane-iteratec) [reply fusionbackend/src/main/java/io/fusion/fusionbackend/service/ModelRepoSyncService.java:99]: Ok

File: fusionfrontend/src/app/core/services/api/if-api.service.ts (hunk 3 above)
Comment (davidlane-iteratec) [reply fusionfrontend/src/app/core/services/api/if-api.service.ts:35]: Ok

**Pull request merged**